libraries/ESP_I2S/src/ESP_I2S.cpp

```diff
@@ -130,45 +232,229 @@ void I2SClass::setInverted(bool bclk, bool ws, bool mclk){
+        case I2S_MODE_PDM_RX:
+        {
+            i2s_pdm_rx_config_t i2s_pdf_rx_config = I2S_PDM_RX_CHAN_CFG(rate, bits_cfg, ch);
+            if (rx_chan != NULL) {
+                rx_sample_rate = rate;
+                rx_data_bit_width = bits_cfg;
+                rx_slot_mode = ch;
+                I2S_ERROR_CHECK_RETURN_FALSE(i2s_channel_init_pdm_rx_mode(rx_chan, &i2s_pdf_rx_config));
+                I2S_ERROR_CHECK_RETURN_FALSE(i2s_channel_enable(rx_chan));
+            }
+            break;
+        }
+    }
+
+    switch (mode){
+        case I2S_MODE_STD:
+            if(!perimanSetPinBus(_mclk, ESP32_BUS_TYPE_I2S_STD, (void *)(this))){ goto err; }
+            if(!perimanSetPinBus(_bclk, ESP32_BUS_TYPE_I2S_STD, (void *)(this))){ goto err; }
+            if(!perimanSetPinBus(_ws,  ESP32_BUS_TYPE_I2S_STD, (void *)(this))){ goto err; }
+            if(_dout >= 0){
+                if(!perimanSetPinBus(_dout,  ESP32_BUS_TYPE_I2S_STD, (void *)(this))){ goto err; }
+            }
+            if(_din >= 0){
+                if(!perimanSetPinBus(_din,  ESP32_BUS_TYPE_I2S_STD, (void *)(this))){ goto err; }
+            }
+            break;
+        case I2S_MODE_TDM:
+            if(!perimanSetPinBus(_mclk, ESP32_BUS_TYPE_I2S_TDM, (void *)(this))){ goto err; }
+            if(!perimanSetPinBus(_bclk, ESP32_BUS_TYPE_I2S_TDM, (void *)(this))){ goto err; }
+            if(!perimanSetPinBus(_ws,  ESP32_BUS_TYPE_I2S_TDM, (void *)(this))){ goto err; }
+            if(_dout >= 0){
+                if(!perimanSetPinBus(_dout,  ESP32_BUS_TYPE_I2S_TDM, (void *)(this))){ goto err; }
+            }
+            if(_din >= 0){
+                if(!perimanSetPinBus(_din,  ESP32_BUS_TYPE_I2S_TDM, (void *)(this))){ goto err; }
+            }
+            break;
+        case I2S_MODE_PDM_TX:
+            if(!perimanSetPinBus(_tx_clk,   ESP32_BUS_TYPE_I2S_PDM, (void *)(this))){ goto err; }
+            if(_tx_dout0 >= 0){
+                if(!perimanSetPinBus(_tx_dout0, ESP32_BUS_TYPE_I2S_PDM, (void *)(this))){ goto err; }
+            }
+            if(_tx_dout1 >= 0){
+                if(!perimanSetPinBus(_tx_dout1, ESP32_BUS_TYPE_I2S_PDM, (void *)(this))){ goto err; }
+            }
+            break;
+        case I2S_MODE_PDM_RX:
+            if(!perimanSetPinBus(_rx_clk,  ESP32_BUS_TYPE_I2S_PDM, (void *)(this))){ goto err; }
+            if(_rx_din0 >= 0){
+                if(!perimanSetPinBus(_rx_din0, ESP32_BUS_TYPE_I2S_PDM, (void *)(this))){ goto err; }
+            }
+            if(_rx_din1 >= 0){
+                if(!perimanSetPinBus(_rx_din1, ESP32_BUS_TYPE_I2S_PDM, (void *)(this))){ goto err; }
+            }
+            if(_rx_din2 >= 0){
+                if(!perimanSetPinBus(_rx_din2, ESP32_BUS_TYPE_I2S_PDM, (void *)(this))){ goto err; }
+            }
+            if(_rx_din3 >= 0){
+                if(!perimanSetPinBus(_rx_din3, ESP32_BUS_TYPE_I2S_PDM, (void *)(this))){ goto err; }
+            }
+            break;
     }
```

Comment:
instead of having 4 cases one after another, have 4 separate private methods to init each type.

Reply:
Done. A lot cleaner code now :)